File: flutteroutfity/flutteroutfity/lib/models/banner.dart
class Banner {
  final String id;
  final String title;
  final String? subtitle;
  final String? description;
  final BannerImage? image;
  final String? link;
  final int position;
  final bool isActive;
  final String? category;

  Banner({
    required this.id,
    required this.title,
    this.subtitle,
    this.description,
    this.image,
    this.link,
    this.position = 0,
    this.isActive = true,
    this.category,
  });

  factory Banner.fromJson(Map<String, dynamic> json) {
    return Banner(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      subtitle: json['subtitle'],
      description: json['description'],
      image: json['image'] != null ? BannerImage.fromJson(json['image']) : null,
      link: json['link'],
      position: json['position'] ?? 0,
      isActive: json['isActive'] ?? true,
      category: json['category'],
    );
  }
}

class BannerImage {
  final String url;
  final String? publicId;

  BannerImage({required this.url, this.publicId});

  factory BannerImage.fromJson(Map<String, dynamic> json) {
    return BannerImage(
      url: json['url'] ?? '',
      publicId: json['public_id'],
    );
  }
}

File: backend/models/Banner.js
const mongoose = require('mongoose');

const bannerSchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, 'Please enter banner title'],
    trim: true,
    maxlength: [100, 'Title cannot exceed 100 characters'],
  },
  subtitle: {
    type: String,
    trim: true,
  },
  description: {
    type: String,
  },
  image: {
    public_id: {
      type: String,
      required: [true, 'Banner image is required'],
    },
    url: {
      type: String,
      required: [true, 'Banner image URL is required'],
    },
  },
  link: {
    type: String,
  },
  position: {
    type: Number,
    default: 0,
  },
  isActive: {
    type: Boolean,
    default: true,
  },
  category: {
    type: String,
    enum: ['men', 'women', 'kids', 'accessories', 'sale', 'general'],
    default: 'general',
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Banner', bannerSchema);
